src/Regex-Core/RxMatcher.class.st

```diff
@@ -113,7 +116,10 @@ RxMatcher >> atEnd [
 { #category : 'testing' }
 RxMatcher >> atEndOfLine [
 
-	^self atEnd or: [stream peek = Cr]
+	^stream peek 
+		ifNotNil: [ :c | c = Cr or: [ c = Lf] ]
```

Comment:
It does not what the method says. In a CRLF environment being on the LF it will report true because there is the CR before

Reply:
For me, the meaning of `#atEndOfLine` does not imply how to deal with a combination of line endings like `#crlf`. I agree that the semantics of this method should never miss the very first line ending after normal characters, because the way how RxMatcher is implemented expects that. But for me this does not imply that "if we have a line ending starting point with CR immediately before a LF, we consider this LF not to be #atEndOfLine". The question for me is, how the result of `#atEndOfLine` is used internally, and I tried to give the safest answer as possible without exploring to the deep. So my proposed approach assumes that a LF should never be considered as something that is **not** a line ending, even though that might lead to fake empty lines.

Again I looked at usage of RX based replacments as we can find in the test case `RxMatcherTest>>#testEmptyStringAtEndOfLine`. The test uses `String cr` for line ending, but the original implementation of RxMatcher also works if you replace it with `String crlf`: The search pattern '$' correctly matches 2 empty strings (same as for `String cr`), thus it just adds the '*' before each position where the line endings were found. But the old implementation of course fails if you use `String lf`.

My proposed PR works for both single char line endings, but the test case fails for `String crlf` because it creates an additional empty line as result. Thus I propose the following refinement:

```
atEndOfLine

	^stream peek 
		ifNotNil: [ :c | c = Cr or: [ c = Lf and: [ self lastChar ~= Cr ] ] ]
		ifNil: [ true ]
```
Please tell me if you would expect the behaviour like that, then I will issue a new PR.

I will also extend `#testEmptyStringAtEndOfLine` to a loop running with all 3 endings.